dot/types/babe.go

```diff
@@ -100,13 +101,17 @@ type ConfigData struct {
 
 // GetSlotFromHeader returns the BABE slot from the given header
 func GetSlotFromHeader(header *Header) (uint64, error) {
-	if len(header.Digest) == 0 {
+	if len(header.Digest.Types) == 0 {
 		return 0, fmt.Errorf("chain head missing digest")
 	}
 
-	digestItem := header.Digest[0]
-	preDigest, ok := digestItem.(*PreRuntimeDigest)
-	if !ok {
+	digestItem := header.Digest.Types[0]
+
+	var preDigest *PreRuntimeDigest
```

Comment:
You can use alternate typecast syntax here if you're looking for a specific type.